src/Databases/DatabaseOverlay.cpp

```diff
@@ -214,38 +277,59 @@ String DatabaseOverlay::getTableDataPath(const ASTCreateQuery & query) const
     return result;
 }
 
+bool DatabaseOverlay::isReadOnly() const
+{
+    return readonly;
+}
+
 UUID DatabaseOverlay::getUUID() const
 {
     UUID result = UUIDHelpers::Nil;
-    for (const auto & db : databases)
+    if (!readonly)
```

Comment:
Let's also add a comment why we expose UUID for non-readonly - in case of non-readonly database (from clickhouse-local) the underlying database is not registered as a separate database, so we let's pretend that overlay database is that underlying database